crates/secrets/src/lib.rs

```diff
@@ -80,17 +82,21 @@ pub enum SecretStoreType {
     #[cfg(feature = "keyring-secret-store")]
     Keyring,
     Kubernetes,
+    #[cfg(feature = "aws-secret-store")]
+    AwsSecretStore,
 }
 
 #[must_use]
+#[allow(unreachable_patterns)]
```

Comment:
I don't think this is right.   What's the reasoning here?

Reply:
Updated, thank you 
```
#[cfg(not(all(feature = "keyring-secret-store", feature = "aws-secrets-manager")))]
 _ => None,
```